CMakeLists.txt

```diff
@@ -128,58 +178,204 @@ set(IO_SOURCE
+)
+target_include_directories(openPMD.io PUBLIC
+    $<BUILD_INTERFACE:${openPMD_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${openPMD_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+if(TARGET Boost::filesystem)
+    target_link_libraries(openPMD.core PUBLIC
+        Boost::boost Boost::system Boost::filesystem)
+    target_link_libraries(openPMD.io PUBLIC
+        Boost::boost Boost::system Boost::filesystem)
+else()
+    target_link_libraries(openPMD.core PUBLIC
+        ${Boost_LIBRARIES})
+    target_link_libraries(openPMD.io PUBLIC
+        ${Boost_LIBRARIES})
+    target_include_directories(openPMD.core SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
+
+    target_include_directories(openPMD.io SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
+endif()
+
+if(openPMD_HAVE_MPI)
+    # MPI targets: CMake 3.9+
+    # note: often the PUBLIC dependency to CXX is missing in C targets...
+    target_link_libraries(openPMD.core PUBLIC MPI::MPI_C MPI::MPI_CXX)
+    target_link_libraries(openPMD.io PUBLIC MPI::MPI_C MPI::MPI_CXX)
+
+    target_compile_definitions(openPMD.core PUBLIC "-DLIBOPENPMD_WITH_MPI=ON")
+    target_compile_definitions(openPMD.io PUBLIC "-DLIBOPENPMD_WITH_MPI=ON")
+else()
+    target_compile_definitions(openPMD.core PUBLIC "-D_NOMPI=ON")
+    target_compile_definitions(openPMD.io PUBLIC "-D_NOMPI=ON")
+endif()
+
+if(openPMD_HAVE_HDF5)
+    target_link_libraries(openPMD.io PUBLIC ${HDF5_LIBRARIES})
+    target_include_directories(openPMD.io SYSTEM PUBLIC ${HDF5_INCLUDE_DIRS})
+    target_compile_definitions(openPMD.io PUBLIC ${HDF5_DEFINITIONS})
+    target_compile_definitions(openPMD.io PRIVATE "-DLIBOPENPMD_WITH_HDF5=ON")
+    if(openPMD_HAVE_MPI)
+        # TODO: remove and just rely on "WITH_MPI"
+        target_compile_definitions(openPMD.io PUBLIC "-DLIBOPENPMD_WITH_PARALLEL_HDF5=ON")
+    endif()
+endif()
+
+if(openPMD_HAVE_ADIOS1)
+    target_link_libraries(openPMD.io ${ADIOS_LIBRARIES})
+    target_include_directories(openPMD.io SYSTEM PUBLIC ${ADIOS_INCLUDE_DIRS})
+    target_compile_definitions(openPMD.io PRIVATE "-DLIBOPENPMD_WITH_ADIOS1=ON")
+    if(openPMD_HAVE_MPI)
+        # TODO: remove and just rely on "WITH_MPI"
+        target_compile_definitions(openPMD.io PUBLIC "-DLIBOPENPMD_WITH_PARALLEL_ADIOS1=ON")
+    endif()
+endif()
+
+if(openPMD_HAVE_ADIOS2)
+    target_link_libraries(openPMD.io PUBLIC ADIOS2::ADIOS2)
+    target_compile_definitions(openPMD.io PRIVATE "-DLIBOPENPMD_WITH_ADIOS2=ON")
+    if(openPMD_HAVE_MPI)
+        # TODO: remove and just rely on "WITH_MPI"
+        target_compile_definitions(openPMD.io PUBLIC "-DLIBOPENPMD_WITH_PARALLEL_ADIOS2=ON")
+    endif()
+endif()
+
+# tests
+set(openPMD_TEST_NAMES
+    openPMDCore
+    openPMDAuxiliary
+    openPMDSerialIO
+#    openPMDParallelIO # TODO: Broken! Does not compile
+)
+# examples
+set(openPMD_EXAMPLE_NAMES
+    poc_HDF5Writer
+    poc_HDF5Reader
+)
+foreach(testname ${openPMD_TEST_NAMES})
+    add_executable(${testname}Tests ${CORE_TESTS})
```

Comment:
This is causing what you interpret as all tests failing. openPMDCoreTests has one failing assertion. But *all* tests are built as CoreTests.

Reply:
Quick hack on this (old.txt = CMakeLists.txt from this PR)

```patch
--- old.txt     2018-01-16 06:28:56.593368093 +0100
+++ new.txt     2018-01-16 06:31:15.554310774 +0100
@@ -178,6 +178,12 @@
         src/IO/HDF5/ParallelHDF5IOHandler.cpp)
 set(CORE_TESTS
         test/CoreTest.cpp)
+set(AUXILIARY_TESTS
+        test/AuxiliaryTest.cpp)
+set(SERIALIO_TESTS
+        test/SerialIOTest.cpp)
+set(PARALLELIO_TESTS
+        test/ParallelIOTest.cpp)

 # library
 add_library(openPMD.core ${CORE_SOURCE})
@@ -266,9 +272,10 @@
     poc_HDF5Reader
 )
 foreach(testname ${openPMD_TEST_NAMES})
-    add_executable(${testname}Tests ${CORE_TESTS})
-    target_link_libraries(${testname}Tests PRIVATE
-        openPMD.core openPMD.io Boost::unit_test_framework)
+    string(TOUPPER ${testname} TEST_SOURCES)
+    add_executable(openPMD${testname}Tests ${${TEST_SOURCES}_TESTS})
+    target_link_libraries(openPMD${testname}Tests PRIVATE
+            openPMD.core openPMD.io Boost::unit_test_framework)
 endforeach()
 foreach(examplename ${openPMD_EXAMPLE_NAMES})
     add_executable(${examplename} ${CORE_TESTS})
@@ -340,12 +347,12 @@
     if(${testname} MATCHES "^Parallel.*$")
         if(openPMD_HAVE_MPI)
             add_test(NAME MPI.${testname}
-                COMMAND ${MPI_TEST_EXE} ${testname}Tests
+                COMMAND ${MPI_TEST_EXE} openPMD${testname}Tests
             )
         endif()
     else()
         add_test(NAME Serial.${testname}
-            COMMAND ${testname}Tests
+            COMMAND openPMD${testname}Tests
         )
     endif()
 endforeach()
```

Reply:
ah yes, I will update the sources. They of course should take the right ones.

So it's ParallelTest and CoreTest that are not compiling right now.